### ISSUE TITLE
geth_heco V1.2.0 can't syning...

### ISSUE THREAD

**Issue by 1b1og-com** [stars-labs/heco-chain#49]
#### System information

Geth version: `geth version`
OS & Version: Windows/Linux/OSX
Commit hash : (if `develop`)

#### Expected behaviour

>  eth_blocknumber is `8523412`

```
INFO [09-25|01:37:56.483] metric                                   method=executeBlock hash=0x14cf09ca3178fced3265e9d68782ac2f7632cf8a4473126daaf19cf4f027d96e number=8523413 trieHash=0s trieProc=0s size=21.49KiB txCount=60 ga
sUsed=5191509 cost=29.910005ms                                                                                                                                                                                                   
ERROR[09-25|01:37:56.484]                                                                                                                                                                                                        
########## BAD BLOCK #########                                                                                                                                                                                                   
Chain config: {ChainID: 128 Homestead: 0 DAO: <nil> DAOSupport: true EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Petersburg: 0 Istanbul: 0, Muir Glacier: <nil>, YOLO v1: <nil>, RedCoastBlock: 6618800, Engine:
 congress}                                                                                                                                                                                                                       
                                                                                                                                                                                                                                 
Number: 8523413                                                                                                                                                                                                                  
Hash: 0x14cf09ca3178fced3265e9d68782ac2f7632cf8a4473126daaf19cf4f027d96e                                                                                                                                                         
         0: cumulative: 22383 gas: 22383 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xedab6537045b91f822618c8efa7c2b56334e3b43d9ebf827b623dd6c69f4937f logs: [0xc0021478c0] bloom: 0000000000000000004000
000000000000000000000000000000000000000000000000000000000000000000000000000020000004000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000001
008000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000800000040000000000000000200000000000000000000000000000000000000000004000000000000000000000000000000000000
0000000000000000000000000000000000000000 state:                                                                                                                                                                                  
         1: cumulative: 59754 gas: 37371 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x21808ad063b18862091b1c0a486ce0e059cb6a51e2c347e7ee91a2d702f6ff84 logs: [0xc002147970] bloom: 0000000000000000000000
000000000000000000000000000000000000000000000000000000002000010000000000000000000004001000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000001
008000000000000000000000000000000000000000000000000020000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000100000000000
0000000000000000000000000000000000000000 state:                                                                                                                                                                                  
         2: cumulative: 474631 gas: 414877 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xb2ce66f8887a995e45f49eaa7c84dc19f3d2f9946ef46ff2ca26e245ee2419ab logs: [0xc002147a20 0xc002147ad0 0xc002147b80] b
loom: 080400000000000000010000000000000000000000000000000000001010000000000000200000000000080000000000400000000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000200000
000000000000000000000000000000200000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000000400000000800000000000800000000000000000800020000000000000000000000000000800000000000
000000012000000000000000100000000000000000000400000000a0000000000000 state:                                                                                                                                                      
         3: cumulative: 790694 gas: 316063 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xb22e60bd3266ccf6ed625b8ae7abbeaff5fba7da007ce371cac5ae5e7448cfc2 logs: [0xc002147c30 0xc002147ce0 0xc002147d90] b
loom: 080000000000000000010000000000000000000000000000000000000010000000000000200000000000000000000000400000000000000000000000000000000000000000000000020000000000000000001400000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000200000000000000000000002000000000040000000000000000000000000000000000000000000000000000000000000000000400000000800000000000800000000000000000800020000000000000000000000000000800000000000
000000010020000000000000100000000000000000000400000000a0000000000000 state:                                                                                                                                                      
         4: cumulative: 1190571 gas: 399877 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x92150d208fd7d452093eee21df836d09215d7c7330767ee7bab964fa021319dc logs: [0xc002147e40 0xc002147ef0 0xc0155a4000] 
bloom: 08000000000000000001000000000000000000000000001000000000001000000000000020000000000000000000000040000000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000010000000000000000000000
000000000000000020000000000000200000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000240000000080000000400080000000000000000080002000000000000000000000000000080000000000
0000000010000000000000000100000000000000000000400000000a0000000000000 state:   

         55: cumulative: 5191509 gas: 64123 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x1826aef1d7deacc72faf580600dfd68b6ac55a2cb8581bec7b1b77fb01cff4f9 logs: [0xc0433286e0] bloom: 0000000000000000000000400000000000000000000000000000000000000000000000000000000000000a000080000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000004800000000000000000000000000000000000000000000000000000000000000000000000000000001020000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000002000000000000000000000000000000000 state:
         56: cumulative: 5191509 gas: 0 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x9b1af769607ef53100668ec9640758575e7e8f012dbfeec0ac08d95b0ca17469 logs: [0xc043328790] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000008008000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000002000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000020000001000001000004000000000000000000000000000000000000000000000000000000000000000000000000 state:
         57: cumulative: 5191509 gas: 0 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x4dd2b1276c4ccf5e78cbe3769144bde99d97c3afeee781ea778421ee4d9c85fa logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state:
         58: cumulative: 5191509 gas: 0 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xbf93bffd4a90b4339b74ba52c17e0f63d055abdf737b74bad866448f6b6dcae6 logs: [0xc043328840] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000010000000000000000000000000000000000000000000002000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000020000101000001000006000000000000000000000000000000000000000000000000000000000000000000000000 state:
         59: cumulative: 5191509 gas: 0 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x46f8aa265060a55e81a9093d562838513c417ed1e119988b1158eba1addc0ff3 logs: [0xc0433288f0] bloom: 00010000000000000000000000040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000084000000000000000000000008008000000000000080000000000000000000080000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000002000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000 state:

Error: invalid merkle root (remote: 1ab173585b61eedaf638e6561191947c9b449f2da534b73bcabdbfd183923990 local: aa4e7f6b497c0b1a496cf308bfbd08c06d557024f294049c5e2c3c67651c659c)
##############################

WARN [09-25|01:39:11.435] Synchronisation failed, dropping peer    peer=7d0add301b6e96bc err="retrieved hash chain is invalid: invalid merkle root (remote: 1ab173585b61eedaf638e6561191947c9b449f2da534b73bcabdbfd183923990 local: aa4e7f6b497c0b1a496cf308bfbd08c06d557024f294049c5e2c3c67651c659c)"
```

**Comment by Nanyan** [comment]
1. Make sure your version IS v1.2.0;
2. Then if you can't sync, it may hitting this [issue 23531](https://github.com/ethereum/go-ethereum/issues/23531)  which is reported in go-ethereum; you can try to fix it by ether: (a) restart your node, and see if it start syncing again; (b) disable the snapshot by setting `--snapshot=false` and then restart your node.

IF you still got problem, please upload more information, including but not limited to the detail version info by `./geth version`; and more logs, etc.

**Comment by allenwq** [comment]
got this problem too, restart helps, but it occurred again after 1 day

**Comment by soltrac** [comment]
I'm having this problem too. Restart helps but sometimes this happens again

**Comment by RooWalletLab** [comment]
I'm having this problem too. Restart helps but sometimes this happens again.first time is 8523412,the second time is 8604671.

**Comment by zzyalbert** [comment]
@allenwq @soltrac @RooWalletLab Yes, it will occur randomly until snapshot generation job is done. After that time, it will not happen again

**Comment by gaiyaa** [comment]
emmm....I'm having this problem too. Restart helps but sometimes this happens again,too

**Comment by Nanyan** [comment]
This was fixed on [release v1.2.2](./../releases/tag/v1.2.2)
